### PR TITLE
feat(ELY-720): Define IAM service consumers

### PR DESCRIPTION
**Pull request by askoriy**

### CODE REVIEW

**Review by sasjo — APPROVED** [review]
LGTM

File: iam/README.md [hunk 1]
```diff
@@ -43,20 +43,23 @@ will be processed independent of others.
...
+|:----------------------------|:-------------------------------------------------------------------------------------------------|:---------|
+| `name`                      | The name of the service system                                                                   | No      |
+| `permission-prefix`         | The permission prefix, this will be prefixed to every permission and roles id. ([a-z][-a-z]{2})  | Yes      |
+| `services`                  | A list of services that need a DAS system                                                        | Yes      |
+| `services.name`             | The service name of the system to be created                                                     | Yes      |
+| `services.repository`       | The description of the service                                                                   | Yes      |
+| `services.allowed-consumers`| A list of clans and services allowed to consume this service                                     | No       |
+| `services.clan`             | Your clan name subscribing to this service                                                       | Yes      |
```
Comment (sasjo) [inline iam/README.md:54]: I think this key needs to be `services.allowed-consumers.clan`. This way to annotate is probably not correct. Maybe it should be something like `services[].allowed-consumers[].clan`, but I guess the first form would clear enough.

File: iam/README.md [hunk 2]
```diff
@@ -43,20 +43,23 @@ will be processed independent of others.
...
+| `services.service-accounts` | A list of service accounts allowed to consume this service. (Full service account name required) | Yes      |
+| `permissions`               | An object containing the permissions for this service. (keys [a-z][-a-z]{1,15})                  | Yes      |
+| `prmissions.<resource>`     | Array containing the verbs for the `<resource>`. (items [a-z][-a-z]{1,15})                       | Yes      |
```
Comment (sasjo) [inline iam/README.md:57]: spelling on `prmissions`

File: iam/README.md [hunk 3]
```diff
@@ -102,3 +105,67 @@ roles:
...
+      - 'quote.list'
+
+```
+
+The consumers will be available in your das system by importing data.consumers. This dataset will contain an array named consumers with service-accounts.
+
+example policy
+
+```
+import data.consumers.consumers
```
Comment (sasjo) [inline iam/README.md:164]: Not a fan of the double consumers... :)

**Review by sasjo — CHANGES_REQUESTED** [review]
Changes LGTM. Some nitty-picky comments.

**Comment by sasjo** [comment]
Are we enforcing alphabetical ordering on the service accounts? If not, we should.

File: iam/README.md [hunk 4]
```diff
@@ -102,3 +105,67 @@ roles:
...
+```yaml
+name: quotes service
+permission-prefix: bhq
+services:
+  - name: braveheart-quotes
+    repository: braveheart-quotes-service
+    allowed-consumers:
+    - clan: iam
+      service-accounts:
+        - 'iam-api@iam-staging-c2a9.iam.gserviceaccount.com'
```
[the remark this answers is not in the document]
Comment (Santos92) [reply iam/README.md]: quotes removed

File: iam/README.md (hunk 1 above)
Comment (Santos92) [reply iam/README.md]: Fixed, full path to property added

File: iam/README.md (hunk 3 above)
Comment (Santos92) [reply iam/README.md]: changed this to data.consumers.services 👍

File: iam/README.md [hunk 5]
```diff
@@ -102,3 +105,67 @@ roles:
...
+    - delete
+    - get
+    - list
+    - update
+roles:
+  - id: admin
+    name: Braveheart quotes administrator
+    desc: full access
+    permissions:
+      - 'quote.create'
```
[the remark this answers is not in the document]
Comment (Santos92) [reply iam/README.md]: done

File: iam/README.md (hunk 2 above)
Comment (Santos92) [reply iam/README.md]: done

**Review by sasjo — APPROVED** [review]

**Pull request merged**